android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -325,11 +339,12 @@ public void onRefresh() {
 
   private void emptyGroupIdQueue() {
     List<PwPair> pwPairs = new ArrayList<>();
+
     for (String groupId : mGroupIdQueue) {
       Log.d(TAG, "groupid " + groupId);
       pwPairs.add(Utils.getTopRankedPwPairByGroupId(mPwCollection, groupId));
```

Comment:
You will need to sort in here, too. (the collection library doesn't know what the best pair in the group is)